### PR TITLE
Do not register LLVMSupport as a library when it should be a component

### PR DESCRIPTION
**Pull request by vgvassilev**
LLVM component must be registered as LLVM_LINK_COMPONENTS to be compatible with LLVM Dylib. Otherwise they are loaded twice in the final binary, once through LLVM Dylmib and once through individual component, and this results in some options being registered twice.

Fix root-project/cling#440

### CODE REVIEW

File: interpreter/cling/tools/driver/CMakeLists.txt [hunk 1]
```diff
@@ -9,9 +9,9 @@
 # Keep symbols for JIT resolution
 set(LLVM_NO_DEAD_STRIP 1)
 
+set(LLVM_LINK_COMPONENTS support)
```
Comment (hahnjo) [inline interpreter/cling/tools/driver/CMakeLists.txt:12]: Should be
```suggestion
set(LLVM_LINK_COMPONENTS Support)
```
I think (looking at upstream LLVM sources)

**Review by hahnjo — COMMENTED** [review]
While we are at it, the Clang libraries should probably be linked with `clang_target_link_libraries` to make `CLANG_LINK_CLANG_DYLIB` work.

**Comment by vgvassilev** [comment]
cc: @serge-sans-paille

File: interpreter/cling/tools/driver/CMakeLists.txt (hunk 1 above)
Comment (serge-sans-paille) [reply interpreter/cling/tools/driver/CMakeLists.txt:12]: I think they are normalized. But capitalized like this, it's more llvmish, :+1:

**Comment by vgvassilev** [comment]
> While we are at it, the Clang libraries should probably be linked with `clang_target_link_libraries` to make `CLANG_LINK_CLANG_DYLIB` work.

I am not sure I completely understand. Can you suggest a change?

**Comment by hahnjo** [comment]
> > While we are at it, the Clang libraries should probably be linked with `clang_target_link_libraries` to make `CLANG_LINK_CLANG_DYLIB` work.
> 
> I am not sure I completely understand. Can you suggest a change?

I think the first branch should read something like
```cmake
set(LIBS
  clingInterpreter
  clingMetaProcessor
  clingUserInterface
  clingUtils
)
add_cling_executable(cling
  cling.cpp
)
clang_target_link_libraries(cling
  clangFrontendTool
)
```
`clang_target_link_libraries` is a function that automatically "does the right thing" depending on `CLANG_LINK_CLANG_DYLIB`. But we can do that in another PR, if needed.

**Comment by vgvassilev** [comment]
@hahnjo how do we move forward here?

**Review by hahnjo — APPROVED** [review]
We can merge (with squash) this change for now, and tackle the other problem later when / if needed.

**Pull request merged**